### PR TITLE
feat: Support custom theme (ref #33)

### PR DESCRIPTION
**Pull request by dongsik-yoo**
### Please check if the PR fulfills these requirements
- [x] It's submitted to right branch according to our branching model
- [x] It's right issue type on title
- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [x] Docs have been added/updated (for bug fixes/features)
- [x] It does not introduce a breaking change or has description for the breaking change

### Description
Supports custom theme at last!

#### The new APIs is `Calendar.setTheme()` which is
```js
/**
 * Set a theme. Using not predefined keys will be returned.
 * @param {object} theme - multiple styles map
 * @returns {Array.<string>} keys - error keys not predefined.
 * @example
 * cal.setTheme({
    'month.dayname.height': '31px',
    'month.dayname.borderTop': '1px solid #e5e5e5',
    'month.dayname.borderBottom': '1px solid #e5e5e5',
 * });
 */
Calendar.prototype.setTheme = function(theme) {
    return this.controller.setTheme(theme);
};
```

#### All theme configurations
```js
var themeConfig = {
    'common.border': '1px solid #e5e5e5',
    'common.backgroundColor': 'white',
    'common.holiday.color': '#ff4040',
    'common.saturday.color': '#135de6',
    'common.dayname.color': '#333',
    'common.today.color': '#135de6',

    // creation guide style
    'common.creationGuide.backgroundColor': 'rgba(81, 92, 230, 0.05)',
    'common.creationGuide.border': '1px solid #515ce6',

    // month header 'dayname'
    'month.dayname.height': '31px',
    'month.dayname.borderTop': '1px solid #e5e5e5',
    'month.dayname.borderLeft': '1px solid #e5e5e5',
    'month.dayname.paddingLeft': '10px',
    'month.dayname.paddingRight': '10px',
    'month.dayname.backgroundColor': 'inherit',
    'month.dayname.fontSize': '12px',
    'month.dayname.fontWeight': 'normal',
    'month.dayname.textAlign': 'left',

    // month day grid cell 'day'
    'month.holidayExceptThisMonth.color': 'rgba(255, 64, 64, 0.4)',
    'month.dayExceptThisMonth.color': 'rgba(51, 51, 51, 0.4)',
    'month.weekend.backgroundColor': 'inherit',
    'month.day.fontSize': '14px',

    // month schedule style
    'month.schedule.borderRadius': '2px',
    'month.schedule.height': '24px',
    'month.schedule.marginTop': '2px',
    'month.schedule.marginLeft': '8px',
    'month.schedule.marginRight': '8px',

    // week header 'dayname'
    'week.dayname.height': '42px',
    'week.dayname.borderTop': '1px solid #e5e5e5',
    'week.dayname.borderBottom': '1px solid #e5e5e5',
    'week.dayname.borderLeft': 'inherit',
    'week.dayname.paddingLeft': '0',
    'week.dayname.backgroundColor': 'inherit',
    'week.dayname.textAlign': 'left',
    'week.today.color': '#333',

    // week vertical panel 'vpanel'
    'week.vpanelSplitter.border': '1px solid #e5e5e5',
    'week.vpanelSplitter.height': '3px',

    // week daygrid 'daygrid'
    'week.daygrid.borderRight': '1px solid #e5e5e5',
    'week.daygrid.backgroundColor': 'inherit',

    'week.daygridLeft.width': '72px',
    'week.daygridLeft.backgroundColor': 'inherit',
    'week.daygridLeft.paddingRight': '8px',
    'week.daygridLeft.borderRight': '1px solid #e5e5e5',

    'week.today.backgroundColor': 'rgba(81, 92, 230, 0.05)',
    'week.weekend.backgroundColor': 'inherit',

    // week timegrid 'timegrid'
    'week.timegridLeft.width': '72px',
    'week.timegridLeft.backgroundColor': 'inherit',
    'week.timegridLeft.borderRight': '1px solid #e5e5e5',
    'week.timegridLeft.fontSize': '11px',

    'week.timegridOneHour.height': '52px',
    'week.timegridHalfHour.height': '26px',
    'week.timegridHalfHour.borderBottom': 'none',
    'week.timegridHorizontalLine.borderBottom': '1px solid #e5e5e5',

    'week.timegrid.paddingRight': '8px',
    'week.timegrid.borderRight': '1px solid #e5e5e5',
    'week.timegridSchedule.borderRadius': '2px',
    'week.timegridSchedule.paddingLeft': '2px',

    'week.currentTime.color': '#515ce6',
    'week.currentTime.fontSize': '11px',
    'week.currentTime.fontWeight': 'normal',
    'week.currentTimeLinePast.border': '1px dashed #515ce6',
    'week.currentTimeLineBullet.backgroundColor': '#515ce6',
    'week.currentTimeLineToday.border': '1px solid #515ce6',
    'week.currentTimeLineFuture.border': 'none',

    // week creation guide style
    'week.creationGuide.color': '#515ce6',
    'week.creationGuide.fontSize': '11px',
    'week.creationGuide.fontWeight': 'bold',

    // week daygrid schedule style
    'week.dayGridSchedule.borderRadius': '2px',
    'week.dayGridSchedule.height': '24px',
    'week.dayGridSchedule.marginTop': '2px',
    'week.dayGridSchedule.marginLeft': '8px',
    'week.dayGridSchedule.marginRight': '8px'
};
```

The prepared themes are two(Default theme and Dooray! theme).

---
Thank you for your contribution to TOAST UI product. 🎉 😘 ✨

### CODE REVIEW

**Comment by dongsik-yoo** [comment]
That build failure was for jenkins configuration.

File: src/js/view/weekday.js [hunk 1]
```diff
@@ -161,4 +162,52 @@ Weekday.prototype._initExceedDate = function(range) {
...
+ * @returns {string} style - color style
+ */
+Weekday.prototype._getDayBackgroundColor = function(theme, day) {
+    var color = '';
+
+    if (theme) {
+        if (day === 0) {
+            color = theme.month.weekend.backgroundColor;
+        } else if (day === 6) {
+            color = theme.month.weekend.backgroundColor;
```
Comment (youjung-hong) [inline src/js/view/weekday.js]: this could changed to short expression, `(day === 0 || day === 6)`

File: test/fixtures/theme-dooray.json [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    "week.currentTimeLinePast.border": "1px solid rgba(19, 93, 230, 0.3)",
+    "week.currentTimeLineBullet.backgroundColor": "#135de6",
+    "week.currentTimeLineToday.border": "1px solid #135de6",
+    "week.currentTimeLineFuture.border": "1px solid #135de6",
+
+    "week.creationGuide.color": "#135de6",
+    "week.creationGuide.fontSize": "12px",
+    "week.creationGuide.fontWeight": "bold",
+
+    "week.dayGridSchedule.borderRadius": "0px",
```
Comment (youjung-hong) [inline test/fixtures/theme-dooray.json]: `0`

**Comment by dongsik-yoo** [comment]
I fixed all your code reviews. Thanks a lot.

**Pull request merged**